WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt

```diff
@@ -90,11 +88,34 @@ class ReaderInterestsViewModel @Inject constructor(
         _uiState.value = uiState
     }
 
-    data class UiState(
+    sealed class UiState(
         val interestsUiState: List<InterestUiState>,
         val interests: ReaderTagList,
-        val doneButtonUiState: DoneButtonUiState
+        val doneButtonUiState: DoneButtonUiState,
+        val progressBarVisible: Boolean = false,
+        val titleVisible: Boolean = false,
+        val subtitleVisible: Boolean = false
     ) {
+        object InitialUiState : UiState(
+            interestsUiState = emptyList(),
+            interests = ReaderTagList(),
+            doneButtonUiState = DoneButtonHiddenUiState,
+            progressBarVisible = true
+        )
+
+        data class ContentLoadSuccessUiState(
+            val interestTagsUiState: List<InterestUiState>,
```

Comment:
I think we can use the `override` keyword so we don't need to pass the values into the constructor. The current implementation creates two separate fields instead of using the field defined in the parent. Wdyt?

Reply:
Using @override is a good idea, thanks!

Also I was thinking to remove interests related fields from the parent and include only in the `ContentLoadSuccessUIState` https://github.com/wordpress-mobile/WordPress-Android/pull/12113#discussion_r440124637

Reply:
👍 sounds good, thanks!